tka/tailchonk.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tka
+
+import (
+	"os"
+	"sync"
+)
+
+// Chonk implementations provide durable storage for AUMs and other
+// TKA state.
+//
+// All methods must be thread-safe.
+//
+// The name 'tailchonk' was coined by @catzkorn.
+type Chonk interface {
+	// GetAUM returns the AUM with the specified digest.
+	//
+	// If the AUM does not exist, then os.ErrNotExist is returned.
+	GetAUM(hash AUMHash) (AUM, error)
+
+	// GetChildAUMs returns all AUMs with a specified previous
+	// AUM hash.
+	GetChildAUMs(prevAUMHash AUMHash) ([]AUM, error)
```

Comment:
If the AUM does not exist does this return a particular error, or an empty slice?

Reply:
Neither! Theres no requirement to know the parent AUM, as the 'child' AUMs all reference the parent hash.